mirror-service/src/test/java/org/acme/client/OsvClientTest.java

```diff
@@ -0,0 +1,97 @@
+package org.acme.client;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.common.ResourceArg;
+import io.quarkus.test.junit.QuarkusTest;
+import org.acme.osv.OsvAnalyzer;
+import org.acme.util.WireMockTestResource;
+import org.acme.util.WireMockTestResource.InjectWireMock;
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import javax.inject.Inject;
+import javax.ws.rs.WebApplicationException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.anyUrl;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.http.Body.fromJsonBytes;
+import static org.acme.util.FileUtil.deleteFileAndDir;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+@QuarkusTest
+@QuarkusTestResource(
+        value = WireMockTestResource.class,
+        initArgs = @ResourceArg(name = "serverUrlProperty", value = "mirror.osv.base.url")
+)
+class OsvClientTest {
+
+    @InjectWireMock
+    WireMockServer wireMockServer;
+
+    @Inject
+    OsvClient osvClient;
+
+    @Inject
+    OsvAnalyzer osvAnalyzer;
+
+    @AfterEach
+    void afterEach() {
+        wireMockServer.resetAll();
+    }
+
+    @Test
+    void testDownloadOsvZips() throws Exception {
+        // Have the OSV API return one zip with one file for the provided ecosystem
+        wireMockServer.stubFor(get(anyUrl())
+                .willReturn(aResponse()
+                        .withStatus(HttpStatus.SC_OK)
+                        .withResponseBody(fromJsonBytes(getOsvTestZipData("osv-vuln.json")))));
```

Comment:
`fromJsonBytes` looks odd given this is a ZIP file. Is there an alternative method that could be used?